bundles/binding/org.openhab.binding.denon/src/main/java/org/openhab/binding/denon/internal/DenonBinding.java

```diff
@@ -137,73 +137,75 @@ protected void removeBindingProvider(DenonBindingProvider bindingProvider) {
      */
     @Override
     public void updated(Dictionary<String, ?> config) throws ConfigurationException {
-        if (config != null) {
-            logger.debug("Denon binding updated");
+        logger.debug("Denon binding updated");
 
-            Enumeration<String> keys = config.keys();
+        if (config == null) {
+            logger.warn("Denon binding has no configuration defined. Check the configuration file.");
+            return;
+        }
 
-            while (keys.hasMoreElements()) {
+        Enumeration<String> keys = config.keys();
 
-                String key = keys.nextElement();
-                if (CONFIG_SERVICE_PID.equals(key)) {
-                    continue;
-                }
+        while (keys.hasMoreElements()) {
+            String key = keys.nextElement();
+            if (CONFIG_SERVICE_PID.equals(key)) {
+                continue;
+            }
 
-                String[] parts = key.split("\\.");
-                String value = ((String) config.get(key)).trim();
+            String[] parts = key.split("\\.");
+            String value = ((String) config.get(key)).trim();
```

Comment:
It's possible that the configuration can have non-String values, so instead of `(String) config.get(key)`, you could safely use `Objects.toString(config.get(key), null)`.

Also, is `trim` really needed?  If the code reading a .cfg file uses [this](https://docs.oracle.com/javase/7/docs/api/java/util/Properties.html#load(java.io.Reader)), trimming the string shouldn't be needed.